src/lib/query.ts

```diff
@@ -18,6 +18,24 @@ export class Query extends BaseQuery {
       this._parameters = { ...this._parameters, ...queryObj };
     }
   }
+  // Validate if input is alphanumeric 
+  private isValidAlphanumeric(input: string): boolean {
+    const alphanumericRegex = /^[a-zA-Z0-9_.-]+$/;
+    return alphanumericRegex.test(input);
+  }
+  // Validate if input is a valid regex pattern
+  private isValidRegexPattern(input: string): boolean {
+    try {
+      new RegExp(input);
```

Comment:
@reeshika-h , [Minor] we can directly call RegExp(input) without the new keyword, by avoiding unnecessary object instantiation.

Reply:
okay